packages/api/src/routers/api/alerts.ts

```diff
@@ -71,6 +74,57 @@ const validateGroupBy = async (req, res, next) => {
 };
 
 // Routes
+router.get('/', async (req, res, next) => {
+  try {
+    const teamId = req.user?.team;
+    if (teamId == null) {
+      return res.sendStatus(403);
+    }
+    const alerts = await Alert.find({ team: teamId });
+    const alertHistories: any = {};
+    for (const alert of alerts) {
+      const histories = await AlertHistory.find({ alert: alert._id })
+        .sort({ createdAt: -1 })
+        .limit(20);
+      alertHistories[alert._id.toString()] = histories;
+    }
```

Comment:
this chunk of code can be optimized using promise all to fetch histories concurrently